include/zephyr/posix/unistd.h

```diff
@@ -11,11 +11,6 @@
 #ifdef CONFIG_POSIX_API
```

Comment:
I have no objection to the change itself, but in commit message of
`posix: net: move gethostname() from static inline to normal function`
it would be nice to see explanation why do we need to do this change.

Reply:
Sure - I think it was a compile error. Will try to dig it up.

Reply:
Fixed!